src/compiler/crystal/codegen/llvm_builder_helper.cr

```diff
@@ -34,6 +34,42 @@ module Crystal
       llvm_type(type).const_int(n)
     end
 
+    def size_t(n, program = @program)
+      if program.bits64?
+        int64(n)
+      elsif program.bits32?
+        int32(n)
+      elsif program.bits16?
+        int16(n)
+      else
+        raise "BUG: only 64, 32 and 16-bit sizes are supported"
+      end
+    end
+
+    def size_t
+      if @program.bits64?
+        llvm_context.int64
+      elsif @program.bits32?
+        llvm_context.int32
+      elsif @program.bits16?
+        llvm_context.int16
+      else
+        raise "BUG: only 64, 32 or 16-bit sizes are supported"
+      end
+    end
```

Comment:
TODO: there should be a `Crystal::Program#size_bit_width` that would be used in these helper methods to choose the proper int representation.